keps/sig-node/2033-rootless/README.md

```diff
@@ -0,0 +1,818 @@
+-->
+
+
+- Alpha: Basic support for rootless mode on cgroups v2 hosts.
+
+- Beta: e2e tests coverage.
+  Requires [the cgroup v2 KEP](../20191118-cgroups-v2.md ) to reach Beta or GA.
+
+- GA: Assuming no negative user feedback based on production experience, promote after >= 2 releases in beta.
+  Requires [the cgroup v2 KEP](../20191118-cgroups-v2.md ) to reach GA.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+This feature is new, there is no upgrade path from existing nodes.
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+N/A
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md.
+
+The production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
```

Comment:
Please add the feature gate, as discussed.

Reply:
Done! updated kep.yaml @ehashman

Reply:
Needs to be added here in the doc too.

Reply:
Updated
```md
* **How can this feature be enabled / disabled in a live cluster?**
  - [X] Feature gate (also fill in values in `kep.yaml`): `KubeletInUserNamespace`
  - [ ] Other
    - Describe the mechanism:

Enabling `KubeletInUsernamespace` feature gate does not automatically execute kubelet in a user namespace.
The user namespace has to be created by RootlessKit before running kubelet.
For `kind` usecase, the namespace is provided by Rootless Docker or Rootless Podman (they internally use RootlessKit).

Note that this feature gate does not support separating kubelet's user namespace from the user namespace of other
node components such as CRI.
All the node components has to run in the same user namespace.
```